runtime/lua/vim/treesitter/query.lua

```diff
@@ -244,8 +244,11 @@ local function xor(x, y)
   return (x or y) and not (x and y)
 end
 
-function Query:match_preds(match, pattern, source)
+function Query:match_preds(match, pattern, source, opts)
+  opts = opts or {}
+
   local preds = self.info.patterns[pattern]
+  local ignore_unknown_predicates = opts.ignore_unknown_predicates
```

Comment:
Ignore all of them??? You should rather be able to register directives and not only predicates.

Reply:
I talked with @vigoux on this and we think it makes sense to just ignore unknown predicates that cause side effects... basically any predicate with ! at the end. I have to make that change to this PR.

Reply:
This is no longer relevant. There is now an API to add directives instead.